src/main/java/io/SocketIO.java

```diff
@@ -21,7 +21,7 @@ public String getMessage() throws IOException {
     InputStreamReader isr = new InputStreamReader(clientSocket.getInputStream());
     BufferedReader reader = new BufferedReader(isr);
     String line = reader.readLine();
-    while (!line.isEmpty()) {
+    while ( line != null && !line.isEmpty()) {
```

Comment:
This looks like a bug fix, but I don't see a corresponding test for it. One great practice to adopt is writing a test to accompany every bug fix. Otherwise someone in the future might come here and say "Hmm, `line != null && !line.isEmpty()` looks duplicative, I think this can just be `!line.isEmpty()`", delete the first part, and since no test would catch it the bug would be re-introduced.